### PR TITLE
Pod topology spread constraints documentation

### PR DESCRIPTION
**Pull request by tjmoore4**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change, as applicable?
 - [x] Have you tested your changes on all related environments with successful results, as applicable?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the new behavior (if this is a feature change)?**
This commit adds the corresponding documentation for using "pod topology
spread constraints" with PGO.

**Other information**:
[ch12440]
[documentation for PR 2678](https://github.com/CrunchyData/postgres-operator/pull/2678)

### CODE REVIEW

File: docs/content/tutorial/high-availability.md [hunk 1]
```diff
@@ -391,6 +391,131 @@ spec:
...
+metadata:
+  name: hippo
+spec:
+  image: registry.developers.crunchydata.com/crunchydata/crunchy-postgres-ha:centos8-13.4-0
+  postgresVersion: 13
+  patroni:
+    dynamicConfiguration:
+      postgresql:
+        parameters:
+          shared_preload_libraries: pgaudit.so
```
Comment (cbandy) [inline docs/content/tutorial/high-availability.md:466]: ⛏️ Some of these fields aren't related to the topic of spread constraints. Can we ditch dynamic config, WAL volume, and images?

File: docs/content/architecture/high-availability.md [hunk 2]
```diff
@@ -187,6 +187,18 @@ this documentation.
...
+
+Kubernetes [Pod Topology Spread Constraints](https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/)
+can also help you efficiently schedule your workloads by ensuring your Pods are 
+not scheduled in only one portion of your Kubernetes cluster. By spreading your
+Pods across your Kubernetes cluster among your various failure-domains, such as
+regions, zones, nodes, and other user-defined topology domains, you can achieve
+high availability as well as efficient resource utilization.
+
+For an example of how pod topology spread constraints work with PGO, please see 
+the [high availability tutorial]({{< relref "tutorial/high-availability.md" >}}##pod-topology-spread-constraints).
```
[the remark this answers is not in the document]
Comment (tjmoore4) [reply docs/content/architecture/high-availability.md:200]: Yes, it should be. Thanks for the catch.

File: docs/content/tutorial/high-availability.md [hunk 3]
```diff
@@ -391,6 +391,131 @@ spec:
...
+
+```
+      topologySpreadConstraints:
+      - maxSkew: <integer>
+        topologyKey: <string>
+        whenUnsatisfiable: <string>
+        labelSelector: <object>
+```
+
+where "maxSkew" describes the maximum degree to which Pods can be unevenly distributed, "topologyKey" is the key that defines a topology in the Nodes' Labels, "whenUnsatisfiable" specifies, when "maxSkew" can't be satisfied, what action should be taken and "labelSelector" is used to find matching Pods.
```
[the remark this answers is not in the document]
Comment (tjmoore4) [reply docs/content/tutorial/high-availability.md:410]: That works for me, updated.

File: docs/content/tutorial/connection-pooling.md [hunk 4]
```diff
@@ -207,6 +207,27 @@ spec:
...
+
+For example, since each of of our pgBouncer Pods will have the standard `postgres-operator.crunchydata.com/role: pgbouncer` Label set, we can use this Label when determining the `maxSkew`. In the example below, since we have 3 nodes with a `maxSkew` of 1 and we've set `whenUnsatisfiable` to `ScheduleAnyway`, we should ideally see 1 Pod on each of the nodes, but our Pods can be distributed less evenly if other constraints keep this from happening.
+
+```
+  proxy:
+    pgBouncer:
+      replicas: 3
+      topologySpreadConstraints: 
+        - maxSkew: 1
+          topologyKey: zone
```
[the remark this answers is not in the document]
Comment (tjmoore4) [reply docs/content/tutorial/connection-pooling.md:222]: I can see how that might be confusing. I'll go with the more generic `my-node-label` option to help make it clear it doesn't necessary have to be a zone, etc label.

File: docs/content/tutorial/high-availability.md (hunk 1 above)
Comment (tjmoore4) [reply docs/content/tutorial/high-availability.md:466]: Agree re: dynamic config and WAL volume. I'd like to leave images in place to stay consistent with the rest of the high-availability section.

**Review by cbandy — APPROVED** [review]

File: docs/content/tutorial/connection-pooling.md [hunk 5]
```diff
@@ -207,6 +207,27 @@ spec:
 
 Note that setting a toleration does not necessarily mean that the PgBouncer instances will be assigned to Nodes with those taints. [Tolerations act as a **key**: they allow for you to access Nodes](https://blog.crunchydata.com/blog/kubernetes-pod-tolerations-and-postgresql-deployment-strategies). If you want to ensure that your PgBouncer instances are deployed to specific nodes, you need to combine setting tolerations with node affinity.
 
+### Pod Spread Constraints
+
+Besides using affinity, anti-affinity and tolerations, you can also set [Topology Spread Constraints]() through `spec.proxy.pgBouncer.topologySpreadConstraints`. This attribute follows the Kubernetes standard topology spread contraint layout.
```
[the remark this answers is not in the document]
Comment (tjmoore4) [reply docs/content/tutorial/connection-pooling.md]: Yes, it should. Updated, thanks.

**Pull request merged**